d2-mode.el

```diff
@@ -163,10 +163,10 @@ Argument FILE-NAME the input file."
   (interactive "fFilename: ")
   (let* ((input file-name)
          (output (concat (file-name-sans-extension input) d2-output-format)))
-    (apply #'call-process d2-location nil "*d2*" nil (list input output))
+    (apply #'call-process (shell-quote-argument d2-location) nil "*d2*" nil (append (list input output) d2-flags))
     (if (equal browse t)
-      (progn
-        (d2-browse-file output))
+        (progn
+          (d2-browse-file output))
       (progn
         (display-buffer (find-file-noselect output t))))))
 
```

Comment:
Add the following:

```suggestion
;;;###autoload
(defsubst string-join (strings &optional separator)
  "Join all STRINGS using SEPARATOR.
Optional argument SEPARATOR must be a string, a vector, or a list of
characters; nil stands for the empty string."
  (declare (pure t) (side-effect-free t))
  (mapconcat #'identity strings separator))
```

Reply:
Note: tested this and it seemed to work. Let's get this in and merge to main. After #47 , bump version.

Reply:
I went for `mapconcat` instead, since then I can specify the separator and can also apply the `shell-quote-argument` to all arguments which is good practice. Hope that is fine as well, given it is part of `fns` I hope it will work, let's see what CI says.